plugins/command_cedict.py

```diff
@@ -0,0 +1,139 @@
+import re
+
+import pyrogram
+from pyrogram import Client, filters
+from pyrogram.types import Message, \
+    InlineKeyboardMarkup, \
+    InlineKeyboardButton, \
+    CallbackQuery
+
+from tools.cedict import Cedict, EntryNotFound
+from tools.utils import command_args_to_str
+
+CEDICT_ENTRIES_PER_PAGE = 10
+
+USAGE_MANUAL_TEXT = ("Usage: \n<code>/cedict your query</code>"
+                     "\n\nFor example:\n<code>/cedict China</code>"
+                     "\n\nYou can also use <code>/dic</code> instead of"
+                     " <code>/cedict</code>")
+
+cedict = Cedict()
+
+
+@Client.on_message(filters.command(["cedict", "dict", "dic"])
+                   & filters.incoming
+                   & ~filters.edited)
+async def command_cedict(_, message: Message):
+    """/cedict question - get translation from cedict"""
+    query = command_args_to_str(message.command)
+    if query:
+        first_page = 0
+        try:
+            cedict_response = cedict.get_cedict_result(query,
+                                                       first_page,
+                                                       CEDICT_ENTRIES_PER_PAGE)
+            text = cedict_response.entries_list_to_str()
+            reply_markup = (
+                _make_next_page_button()
+                if cedict_response.has_next_page
+                else None
+            )
+        except EntryNotFound:
+            text = _get_nothing_was_found_message(query)
+            reply_markup = None
+    else:
+        text = USAGE_MANUAL_TEXT
+        reply_markup = None
+
+    await message.reply(text, reply_markup=reply_markup, quote=True)
+
+
+@Client.on_callback_query(filters.regex(
+    r"^cedict_(previous|next)_button_pressed$"
+))
+async def callback_query_cedict_button_pressed(_,
+                                               callback_query: CallbackQuery):
+    current_page_first_entry_number = re.search(
+        '[0-9]+',
+        callback_query.message.text
+    ).group()
```

Comment:
you can use `re.compile` instead of `re.search` for this pattern for match it faster, although you wouldn't feel the difference of the performance

```py
# out of this handler
PAGE_PATTERN = re.compile('[0-9]+')
# inside this handler
current_page_first_entry_number = PAGE_PATTERN.search(
    callback_query.message.text
).group()
```

Reply:
alright i've replaced with compile, but what makes it faster, i'm curious

Reply:
> but using re.compile() and saving the resulting regular expression object for reuse is more efficient when the expression will be used several times in a single program.

https://docs.python.org/3.10/library/re.html#re.compile